src/vcf/fixer.cpp

```diff
@@ -210,10 +243,139 @@ namespace ebi
         }
+
+        size_t Fixer::remove_duplicate_strings(const std::string &column,
+                                               const std::string &separator)
+        {
+            std::set<std::string> already_present;
+            auto is_value_duplicated = [&](const std::string &value, size_t index) -> bool {
+                bool first_appearance = already_present.insert(value).second;
+                return not first_appearance;
+            };
+            return remove_fields(column, separator, is_value_duplicated);
+        }
+
+        size_t Fixer::remove_duplicate_key_value_pairs(const std::string &column,
+                                                       const std::string &separator,
+                                                       const std::string &key_value_separator,
+                                                       const std::string &empty_value)
+        {
+            std::map<std::string, std::string> values;
+            std::vector<std::string> ordered_values;
+            std::set<std::string> fields_to_remove;
+
+            std::vector<std::string> fields;
+            util::string_split(column, separator.c_str(), fields);
+
+            for (auto & field : fields) {
+                std::vector<std::string> subfields;
+                util::string_split(field, key_value_separator.c_str(), subfields);
+                auto iterator = values.find(subfields[0]);
+                if (iterator == values.end()) {
+                    ordered_values.push_back(subfields[0]);
+                    values[subfields[0]] = subfields[1];
+                } else if (values[subfields[0]] != subfields[1]) {
+                    fields_to_remove.insert(subfields[0]);
+                }
+            }
+
+            std::string fixed_column;
+            size_t num_removed_duplicates = 0;
+            for (auto & ordered_value : ordered_values) {
+                if (fields_to_remove.find(ordered_value) == fields_to_remove.end()) {
+                    fixed_column += ordered_value + key_value_separator + values[ordered_value] + separator;
+                } else {
+                    num_removed_duplicates++;
+                }
+            }
+            if (fixed_column.size() == 0) {       // all the fields were removed
+                fixed_column = empty_value;
+            } else {
+                fixed_column.pop_back();          // remove trailing separator (its length is always 1)
+            }
+
+            output << fixed_column;
+            return num_removed_duplicates;
+        }
+
+        size_t Fixer::remove_duplicate_format_sample_pairs(const std::string &string_line) {
+            const size_t format_column_index = 8;
+            size_t num_removed_duplicates = 0;
+
+            using iter = std::vector<std::string>::iterator;
+            fix_columns(format_column_index, -1, string_line, "\t", [&](iter first, iter last) {
```

Comment:
I think @jmmut and @Anishka0107 discussed a way to optimize the loops in this method a bit, right?